### PR TITLE
types: fix evidence timestamp calculation

### LINKED ISSUE #5030: Clarify / fix duplicate vote evidence timestamp calculation
Ref https://github.com/tendermint/tendermint/blob/95d2d136cdcff2a3bb29d4e4271d757e4f88867c/types/evidence.go#L315 - at the moment, it seems like the votes are sorted lexicographically, and the timestamp of the first vote is used as the timestamp of the evidence (e.g. used in validity determination based on the unbonding period) - to me, this seems arbitrary, it seems like we should use some calculation that is not dependent on the lexicographical order of the votes.

### PR DESCRIPTION
**Pull request by melekes**
depending on the votes order in DuplicateVoteEvidence is arbitrary.
we should always choose the latest timestamp.

Closes #5030

### CODE REVIEW

**Comment by cwgoes** [comment]
Should we always use the latest timestamp? Won't that mean that we consider evidence invalid if one vote is within the unbonding period, but the other vote is not? Is that the behaviour we want?

**Comment by melekes** [comment]
> Won't that mean that we consider evidence invalid if one vote is within the unbonding period, but the other vote is not?

by always picking the later timestamp, we increase the chance of evidence being valid.

**Comment by cmwaters** [comment]
I agree with using the latest timestamp. Although both votes are necessary for the evidence  it would make sense if we consider the latest vote to be the one that actually breaks consensus and hence should be the timestamp of that one used for DuplicateVoteEvidence

**Comment by cwgoes** [comment]
> > Won't that mean that we consider evidence invalid if one vote is within the unbonding period, but the other vote is not?
> 
> by always picking the later timestamp, we increase the chance of evidence being valid.

Ah, right, that makes sense.

File: types/evidence.go [hunk 1]
```diff
@@ -1770,3 +1775,10 @@ func NewMockPOLC(height int64, time time.Time, pubKey crypto.PubKey) ProofOfLock
 		PubKey: pubKey,
 	}
 }
+
+func maxTime(t1 time.Time, t2 time.Time) time.Time {
```
Comment (marbar3778) [inline types/evidence.go:1779]: Should this be added to time pkg in libs?

Comment (melekes) [reply types/evidence.go:1779]: not until it's used (copied) in multiple packages

**Pull request merged**